CMake fixes/improvements

Ichthyostega opened the pull request:
...some small bits found out while packaging for Debian/PPA

see https://github.com/NatronGitHub/Natron/issues/215

Disclaimer: I used these tweaks on Debian and Ubuntu Linux. I can not easily verify if these changes break anything on other platforms.

Ichthyostega committed:
CMake: OpenGL GLVND compatibility

Linux systems are gradually switching to the GLVND ABI;
thus we might switch to the compatibility mechanism provided by CMake,
which boils down to using the OPENGL_LIBRARIES variable and to declare
explicitly that we prefer GLVND when available

https://cmake.org/cmake/help/v3.11/policy/CMP0072.html
https://cmake.org/cmake/help/v3.11/module/FindOpenGL.html#linux-specific

Ichthyostega committed:
CMake: fix librevenge dependency

Librevenge has been split into several libraries, and thus we need
to check for all three pkg-config files to pick up all link flags.

Ichthyostega committed:
CMake: include the HaldCLUT plugin into the bundle

Note: building on Linux/Debian encountered some problems with the dependencies
picked up by CMake when using the Debian package libcurl4-gnutls-dev; we got
several link flags for (TLS, SSL) libraries which are already transitive
dependencies of the libcurl-gnutls. To work around these, without having
to build-depend on a ton of further SSL related packages, we include the
library as always linked dynamically (by using ${CURL_LDFLAGS} without
the _STATIC_ infix)

Generally speaking, it would be an extremely bad idea to link libcurl
statically (think of vulnerabilities). However, on Debian, dynamic linking
is generally preferred and CMake links results from pkg-config dynamic by default.

rodlie commented:
Thanks for the fixes and sorry for the late reply, been busy this week.

HaldCLUT is not part of the official Natron bundle so that commit should be reverted, or enhanced to make this a ON/OFF option(?).

Ichthyostega committed:
CMake: change HaldCLUT into an optional component

..as Ole-André Rodlie pointed out in #8,
HaldCLUT is not considered part of the official Natron bundle.
Moreover, it requires libCurl for download of ~100 MiB of CLUT profiles

rodlie commented:
Great, thanks.
